From 10ba14ae37825cea1dc32d556964b6fbec0f2336 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 18 Aug 2022 14:43:58 +0300
Subject: [PATCH] [SW-86869] gaudi3: remove rom recovery call from erom recovery

Currently, in case of erom recovery manifest load or signature
verification failure, ROM executes ROM recovery where PCIe link,
I2C and range registers are disabled based on security_enable
fuse bit.

As part of secure code review, it is recommended to keep ROM
and EROM flows separate and in case of erom recovery failure, ROM
should halt and not enter ROM recovery flow.

Change-Id: I77da7786749c7bb2b9a5d36ce4a8abfe05676ddb
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index b714f9c..429cab7 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -218,19 +218,14 @@
 	count = boot_desc_init(HL_FW_BOOT_PTR_EROM);
 	if (!count) {
 		arcboot_rom_status_update(PROGRESS_INIT_EROM_BOOT_DESC_FAIL);
-		goto rom_recovery;
+		return;
 	}
 
 	if (arcboot_rom_verify(HL_FW_BOOT_PTR_EROM))
-		goto rom_recovery;
+		return;
 
 	for (i = 0; i < count; i++)
 		ext_rom_load_erom_image(i);
-
-	return;
-
-rom_recovery:
-	execute_rom_recovery();
 }
 
 void execute_recovery(void)
